test/writer.markua

```diff
@@ -0,0 +1,718 @@
+
+Greater-than: >
+
+Hash: #
+
+Period: .
+
+Bang: !
+
+Plus: +
+
+Minus: -
+
+* * *
+
+{id: links}
+# Links
+
+{id: explicit}
+## Explicit
+
+Just a [URL](/url/).
+
+[URL and title](/url/){title: "title"}.
+
+[URL and title](/url/){title: "title preceded by two spaces"}.
+
+[URL and title](/url/){title: "title preceded by a tab"}.
+
+[URL and title](/url/){title: "title with "quotes" in it"}
+
+[URL and title](/url/){title: "title with single quotes"}
+
+[with_underscore](/url/with_underscore)
+
+[Email link](mailto:nobody@nowhere.net)
+
+[Empty]().
+
+{id: reference}
+## Reference
+
+Foo [bar](/url/).
+
+With [embedded [brackets]](/url/).
+
+[b](/url/) by itself should be a link.
+
+Indented [once](/url).
+
+Indented [twice](/url).
+
+Indented [thrice](/url).
+
+This should [not][] be a link.
+
+```
+[not]: /url
+```
+
+Foo [bar](/url/){title: "Title with "quotes" inside"}.
+
+Foo [biz](/url/){title: "Title with "quote" inside"}.
+
+{id: with-ampersands}
+## With ampersands
+
+Here’s a [link with an ampersand in the URL](http://example.com/?foo=1&bar=2).
+
+Here’s a link with an amersand in the link text:
+[AT&T](http://att.com/){title: "AT&T"}.
+
+Here’s an [inline link](/script?foo=1&bar=2).
+
+Here’s an [inline link in pointy braces](/script?foo=1&bar=2).
+
+{id: autolinks}
+## Autolinks
+
+With an ampersand:
+[http:~/~/example.com/?foo=1&bar=2](http://example.com/?foo=1&bar=2){class: uri}
+
+* In a list?
+* [http:~/~/example.com/](http://example.com/){class: uri}
+* It should.
+
+An e-mail address: [nobody@nowhere.net](mailto:nobody@nowhere.net){class: email}
+
+> Blockquoted: [http:~/~/example.com/](http://example.com/){class: uri}
+
+Auto-links should not occur here: `<http://example.com/>`
+
+```
+or here: <http://example.com/>
+```
+
+* * *
+
+{id: images}
+# Images
+
+From "Voyage dans la Lune" by Georges Melies (1902):
+
+{alt: "lalune", title: "Voyage dans la Lune"}
+![](lalune.jpg)
+
+Here is a movie
+{alt: "movie"}
+![](movie.jpg)
+icon.
+
+* * *
+
+{id: footnotes}
+# Footnotes
+
+Here is a footnote reference,[^1] and another.[^2] This should *not* be a
+footnote reference, because it contains a space.[^my note] Here is an inline
+note.[^3]
+
+> Notes can go in quotes.[^4]
+
+1. And in list items.[^5]
+
+This paragraph should not be part of the note, as it is not indented.
+
+[^1]: Here is the footnote. It can go anywhere after the footnote reference. It
+    need not be placed at the end of the document.
+
+[^2]: Here’s the long note. This one contains multiple blocks.
+
+    Subsequent blocks are indented to show that they belong to the footnote (as
+    with list items).
+
+    ```
+      { <code> }
+    ```
+
+    If you want, you can indent every line, but you can also be lazy and just
+    indent the first line of each block.
+
+[^3]: This is *easier* to type. Inline notes may contain
+    [links](http://google.com) and `]` verbatim characters, as well as
+    [bracketed text].
+
+[^4]: In quote.
+
+[^5]: In list.
```

Comment:
Have you run this whole test output through Markua and checked to make sure everything looks right when rendered?
(Ditto with the table test?)

Reply:
Unfortunately there is currently no automatic test-tool for Markua, but we manually checked the files again, to make sure that everything is proper Markua.